src/cli/commands/files/add.js

```diff
@@ -19,15 +24,72 @@ module.exports = Command.extend({
   },
 
   run: (recursive, path) => {
-    var node = new IPFS()
-    path = process.cwd() + '/' + path
-    node.files.add(path, {
-      recursive: recursive
-    }, (err, stats) => {
+    let s
+    let rs
+
+    if (!path) {
+      throw new Error('Error: Argument \'path\' is required')
+    }
+
+    s = fs.statSync(path)
+
+    if (s.isDirectory() && recursive === false) {
+      throw new Error('Error: ' + process.cwd() + ' is a directory, use the \'-r\' flag to specify directories')
+    }
+    if (path === '.' && recursive === true) {
+      path = process.cwd()
+      s = fs.statSync(process.cwd())
+    } else if (path === '.' && recursive === false) {
+      s = fs.statSync(process.cwd())
+      if (s.isDirectory()) {
+        throw new Error('Error: ' + process.cwd() + ' is a directory, use the \'-r\' flag to specify directories')
+      }
+    }
```

Comment:
It's to check for all of the inputs you could give to add command. variations i could think of are...

1) "." add the cwd but throw error for no recursion flag
2) "." -r add the cwd
3) "/some/path" but throw error for no recursion
4) "/some/path" -r 
5) No path, throw err
6) filename.type